aspnetcore/security/data-protection/using-data-protection.md

```diff
@@ -29,3 +32,32 @@ When you create a protector you must provide one or more [Purpose Strings](xref:
 > Instances of `IDataProtectionProvider` and `IDataProtector` are thread-safe for multiple callers. It's intended that once a component gets a reference to an `IDataProtector` via a call to `CreateProtector`, it will use that reference for multiple calls to `Protect` and `Unprotect`.
 >
 >A call to `Unprotect` will throw CryptographicException if the protected payload cannot be verified or deciphered. Some components may wish to ignore errors during unprotect operations; a component which reads authentication cookies might handle this error and treat the request as if it had no cookie at all rather than fail the request outright. Components which want this behavior should specifically catch CryptographicException instead of swallowing all exceptions.
+
+Consider the following code which uses a service provider because the implementation of `IXmlRepository` has a dependency on a singleton service:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    // ...
+
+    var sp = services.BuildServiceProvider();
+    services.AddDataProtection()
+      .AddKeyManagementOptions(o => o.XmlRepository = sp.GetService<IXmlRepository>());
+}
+```
+
+The preceding code logs the following warning:
+
+  *Calling 'BuildServiceProvider' from application code results in an additional copy of singleton services being created. Consider alternatives such as dependency injecting services as parameters to 'Configure'.*
+
+The following code provides the `IXmlRepository` implementation without having to build the service provider and therefore making additional copies of singleton services:
+
+```csharp
+services.AddOptions<KeyManagementOptions>()
+        .Configure<IMyDataStore>((options, store) => 
```

Comment:
I'd suggest making it more explicit the dependency passed to this `Configure<>()` is not supposed to be scoped (hence you shouldn't inject a `DbContext`), instead you should inject some factory like `IServiceScopeFactory` and use it in your `IXmlRepository` implementation to create a new scope and get scoped dependencies like `DbContext` if you need.